File: frontend/src/app/models/ProxyDetail.ts
import {ProxyStatistic} from './ProxyStatistic';
import {ProxyReputationBreakdown} from './ProxyReputation';

export interface ProxyDetail {
  id: number;
  ip: string;
  port: number;
  username: string;
  password: string;
  has_auth: boolean;
  estimated_type: string;
  country: string;
  created_at: string;
  latest_check?: string | null;
  latest_statistic?: ProxyStatistic | null;
  reputation?: ProxyReputationBreakdown | null;
}

File: backend/internal/api/dto/proxy_detail.go
package dto

import "time"

type ProxyDetail struct {
	Id              int                       `json:"id"`
	IP              string                    `json:"ip"`
	Port            uint16                    `json:"port"`
	Username        string                    `json:"username"`
	Password        string                    `json:"password"`
	HasAuth         bool                      `json:"has_auth"`
	EstimatedType   string                    `json:"estimated_type"`
	Country         string                    `json:"country"`
	CreatedAt       time.Time                 `json:"created_at"`
	LatestCheck     *time.Time                `json:"latest_check,omitempty"`
	LatestStatistic *ProxyStatistic           `json:"latest_statistic,omitempty"`
	Reputation      *ProxyReputationBreakdown `json:"reputation,omitempty"`
}
